routing/index_graph_starter.cpp

```diff
@@ -76,10 +100,46 @@ void IndexGraphStarter::RedressRoute(vector<Joint::Id> const & route,
   }
 }
 
+void IndexGraphStarter::AddZeroLengthOnewayFeature(Joint::Id from, Joint::Id to)
+{
+  if (from == to)
+    return;
+
+  m_jointsOfAdditionalEdges.insert(from);
+  m_jointsOfAdditionalEdges.insert(to);
+  m_additionalZeroEdges.emplace_back(from, to, m_additionalNextFeatureId++);
+}
+
+void IndexGraphStarter::GetAdditionalEdgesList(Joint::Id jointId, bool isOutgoing,
+                                               vector<JointEdge> & edges)
+{
+  if (m_jointsOfAdditionalEdges.count(jointId) == 0)
+    return;
+
+  for (DirectedEdge const e : m_additionalZeroEdges)
+  {
+    if (isOutgoing)
+    {
+      if (e.GetFrom() == jointId)
+        edges.emplace_back(e.GetTo(), 0 /* weight */);
```

Comment:
You correctly understood what I mean by the word "potential" in the context of routing.